RecoPixelVertexing/PixelLowPtUtilities/plugins/ClusterShapeHitFilterESProducer.cc

```diff
@@ -0,0 +1,98 @@
+
+  const std::string pixelShapeFile;
+  const std::string pixelShapeFileL1;
+  const float minGoodPixelCharge_, minGoodStripCharge_;
+  const bool cutOnPixelCharge_, cutOnStripCharge_;
+  const bool cutOnPixelShape_, cutOnStripShape_;
+};
+
+/*****************************************************************************/
+ClusterShapeHitFilterESProducer::ClusterShapeHitFilterESProducer(const edm::ParameterSet& iConfig)
+    : pixelShapeFile(iConfig.getParameter<std::string>("PixelShapeFile")),
+      pixelShapeFileL1(iConfig.getParameter<std::string>("PixelShapeFileL1")),
+      minGoodPixelCharge_(0),
+      minGoodStripCharge_(clusterChargeCut(iConfig)),
+      cutOnPixelCharge_(false),
+      cutOnStripCharge_(minGoodStripCharge_ > 0),
+      cutOnPixelShape_(iConfig.getParameter<bool>("doPixelShapeCut")),
+      cutOnStripShape_(iConfig.getParameter<bool>("doStripShapeCut")) {
+  std::string componentName = iConfig.getParameter<std::string>("ComponentName");
+
+  edm::LogInfo("ClusterShapeHitFilterESProducer") << " with name: " << componentName;
+
+  setWhatProduced(this, componentName)
+      .setConsumes(fieldToken_)
+      .setConsumes(geoToken_)
+      .setConsumes(topolToken_)
+      .setConsumes(pixelToken_)
+      .setConsumes(stripToken_);
+}
+
+/*****************************************************************************/
+ClusterShapeHitFilterESProducer::ReturnType ClusterShapeHitFilterESProducer::produce(
+    const ClusterShapeHitFilter::Record& iRecord) {
+  using namespace edm::es;
+
+  // Produce the filter using the plugin factory
+  ClusterShapeHitFilterESProducer::ReturnType aFilter(new ClusterShapeHitFilter(&iRecord.get(geoToken_),
+                                                                                &iRecord.get(topolToken_),
+                                                                                &iRecord.get(fieldToken_),
+                                                                                &iRecord.get(pixelToken_),
+                                                                                &iRecord.get(stripToken_),
+                                                                                pixelShapeFile,
+                                                                                pixelShapeFileL1));
+
+  aFilter->setShapeCuts(cutOnPixelShape_, cutOnStripShape_);
+  aFilter->setChargeCuts(cutOnPixelCharge_, minGoodPixelCharge_, cutOnStripCharge_, minGoodStripCharge_);
+  return aFilter;
+}
+
+/*****************************************************************************/
+void ClusterShapeHitFilterESProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<std::string>("PixelShapeFile");
+  desc.add<std::string>("PixelShapeFileL1");
+  desc.add<std::string>("ComponentName");
+  desc.add<bool>("doPixelShapeCut", true);
+  desc.add<bool>("doStripShapeCut", true);
```

Comment:
@perrotta I believe this is also not like #27589. Here the only defaults set are what was previously being set as defaults in the C++ code for tracked parameters by playing the `exists` game:
https://github.com/cms-sw/cmssw/blob/master/RecoPixelVertexing/PixelLowPtUtilities/src/ClusterShapeHitFilterESProducer.cc#L21-L22

So again, using the `fillDescriptions` is the proper way we want people to inject defaults to tracked parameters from within C++.